workspaces/liferay-sample-workspace/client-extensions/liferay-sample-data-set-filter/src/index.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import type {FDSFilter} from '@liferay/js-api/data-set';
+
+const mySampleFilter: FDSFilter = ({filter, setFilter}) => {
+	const div = document.createElement('div');
+	const button = document.createElement('button');
+	const input = document.createElement('input');
+
+	div.className = 'dropdown-item';
+
+	button.className = 'btn btn-block btn-secondary btn-sm mt-2';
+	button.innerText = 'Submit'
+	button.onclick = () => setFilter({
```

Comment:
This is leaving the filter UX the responsibility of controlling the filter applicationl, in line with what current ootb FDS filters do ([example](https://github.com/liferay/liferay-portal/blob/cb3bd2d3d5105aa791a43d967f8eb20620829eaf/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DateRangeFilter.js#L163-L204))

However I'd like to check if this shall be part of the contract, or else, the "apply", "delete" and "edit filter" would instead be managed centrally from the FDS for all filters (ootb included). The idea would be that the CX can tell the FDS when the info is ready to apply the filter (value of `disabled` prop), and when we're adding a filter or editing it. This might sound crazy at a first sight but removes responsibility from the filter creator and ends up providing a more unifed experience for end users.

The cost is that filter contract is a bit more opinionated

Thoughts?

Reply:
> The idea would be that the CX can tell the FDS when the info is ready to apply the filter (value of disabled prop), and when we're adding a filter or editing it.

@dsanz I'm not totally sure what you mean here

I believe the filter creator would have the knowledge of when to "enable" it for a given table, note that this functionality isn't included yet in this example.

I do think `setFilter` is necessary, because we need a way to set both the preSelectedData and the odataString. We couldn't get by with just odataString because we need to pre-populate.

I think this implementation is so far the more simple and straight forward, I would prefer to start with this and then iterate as needed as I think this provides a baseline. When we start to look into "detached" filters and a global API, that will provide more opportunities and escape hatches for developers I think.